### DISCUSSION TITLE
AOC 2022 Day Three

### DISCUSSION THREAD

**Discussion by clandau** [Virtual-Coffee/virtualcoffee.io#757] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 3](https://adventofcode.com/2022/day/3) here

-----

VC Advent of code leaderboard
Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private) and use join code 363496-e6f42283

**Comment by clandau** [comment]
JS: https://github.com/clandau/advent-of-code/blob/main/2022/day-3/index.js

I was extremely excited to get to part 2 and be able to reuse a function.

**Reply by tkshill** [reply]
This looks great! While loops are probably the most efficient way to do the searches I imagine.

**Reply by doleraj** [reply]
Depends on what you've got for libraries - using something like Linq means you've got GroupBy functionality available which was apparently V Fast for some of my coworkers' solutions.

**Reply by danieltott** [reply]
`sack of data` made me laugh out loud 😂

**Reply by clandau** [reply]
yea I probably could have named the parameter 😆.  `sack of sacks`?

**Reply by bogdaaamn** [reply]
Wowwww this is so pretty and clean 😭

I love you went for string indexes (I did them in arrays) but I love yours so much better with the while loop and everything!!

**Comment by doleraj** [comment]
Tried to be at least a LITTLE conscious of runtime today. That let me cheat and use `.includes` instead of trying to assemble letter groups. https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day03/index.ts

**Reply by doleraj** [reply]
Today I also submitted a PR for `aocrunner`, because the README.md generation wasn't working for me. Turns out it was Windows line endings. 👊

**Reply by tkshill** [reply]
There is no greater victory than a PR on some OSS that you use. Also, `includes` and `reduce` solve like, 90% of all my data parsing conundrums. They are the best!

**Reply by bogdaaamn** [reply]
Noice, I was proud of myself using `if (i % 3 !== 0) {` as well. I felt like in school, I don't think I used module in 5 years 😂

**Reply by danieltott** [reply]
Yep - definitely have to look up modulo any time I need it 😂

**Comment by turingtestable** [comment]
Definitely my ugliest solution thus far, but it functions! [ruby](https://github.com/turingtestable/AdventOfCode2022/blob/master/challenge3/solution.rb)

**Reply by tkshill** [reply]
Ruby's ugliest code is still prettier than every other language! It makes everything look so easy! Good stuff!

**Comment by tkshill** [comment]
This is a good question for F#. There's a lot of good built ins in the Collection libraries. 

Answer [here](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day03.fs)

Most of my debugging time was because I fail at basic arithmetic.

**Reply by clandau** [reply]
woah I guess so! nice

**Reply by Cerchie** [reply]
Is that... 14 lines of code? very nice

**Reply by bogdaaamn** [reply]
This is getting insane 😭

**Reply by bogdaaamn** [reply]
(in a good way)

**Comment by davidalpert** [comment]
F# for day 03
- logic: [day03_rucksack_reorganization.fs](https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day03_rucksack_reorganization.fs)
- tests: [day03_tests.fs#L325-L402](https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day03_tests.fs#L325-L402)

I often write tests to drive out and prove my logic against each step of the example; I've caught more bugs that way... 😎

**Reply by davidalpert** [reply]
and as usual, @tkshill's solution is significantly more concise and elegant than mine 😉

**Reply by tkshill** [reply]
I like seeing your F#. Your solution is easier to test, and perhaps expand. You have a real knack for thinking in systems. I'm just lazy and never want to define my own functions (or do any optimisation if the compiler doesn't yell at me) 😂

**Reply by tkshill** [reply]
If I had done the priority logic your way I wouldn't have spent so much time debugging.

**Comment by danimeyer** [comment]
Another ruby: https://github.com/danimeyer/advent-of-code/blob/main/2022/3.rb

I am learning all kinds of ruby methods. I still feel like JavaScript is the language I know best, and it is _wild_ how many methods and ways of doing things exist in ruby!

**Reply by tkshill** [reply]
This is great! Super easy to understand and looks pretty performant

**Reply by danimeyer** [reply]
I just assume my solutions are terrible and definitely don't take advantage of ruby, so thanks for the confidence boost! But I also feel like I'm learning a lot from reading other people's solutions, which is awesome! And the whole point I'm finally doing Advent of Code this year 🎄

**Reply by bogdaaamn** [reply]
I love a good readable ruby solution (I don't know ruby)!! Looking good 🎄

**Comment by danieltott** [comment]
Here's my ts solution! 

https://github.com/danieltott/aoc2022/blob/main/src/day03/index.ts

I feel pretty good about the efficiency of these, but I'm sure that I will feel bad about it again after seeing y'alls solutions 😂

**Reply by clandau** [reply]
ooo ours minds were working the same today :) very similar implementation. nice use of Set!

**Reply by tkshill** [reply]
One day I'm gonna get you writing Haskell or something. 😄
All your code is so functional. This solution looks awesome.

**Reply by doleraj** [reply]
Jesus, sub-millisecond responses for both parts? I'm gonna go crawl under a rock now...

**Reply by bogdaaamn** [reply]
Once again being humbled by Dan's solution 😂 I need to step up my game next week

**Reply by danieltott** [reply]
@doleraj I feel like those timing results are dependent on the machine and everything too. I wonder if there's a nice way to do automated performance tests on results that are built using `aocrunner`

**Comment by Cerchie** [comment]
Again, a just-get-it-done solution but I'm proud of the way I've been taking it step-by-step! 

solution 1: https://github.com/Cerchie/adventofcode2022/blob/main/day-03/index.js

solution 2: https://github.com/Cerchie/adventofcode2022/blob/main/day-03/index2.js

**Reply by tkshill** [reply]
Cool use of filestream. Also, glad you're joining us!

**Reply by bogdaaamn** [reply]
I like it!! Nice that you used the map for the letter priorities. I was building one for myself, but I quickly googled and found out about [charCodeAt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt)

**Comment by bogdaaamn** [comment]
Another day, another reducer https://github.com/BogDAAAMN/advent-of-code-2022/blob/main/day-03.js

Honestly not super proud of these either, but it was nice to learn about [charCodeAt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt). Although it took me a while to understand why it's not working and implement the offset.

**Comment by aishwarya-mali** [comment]
JS: https://github.com/aishwarya-mali/advent-of-code/blob/main/2022/Day-3/script.js
Part 1 was difficult for me.
